src/ConstructiveSolidGeometry/SurfacePrimitives/RegularPrismMantle.jl

```diff
@@ -0,0 +1,55 @@
+struct RegularPrismMantle{N,T,TZ} <: AbstractSurfacePrimitive{T}
+    r::T
+    z::TZ
+    function RegularPrismMantle( N::Integer,
+                   ::Type{T},
+                   r::T,
+                   z::Union{T, <:AbstractInterval{T}}) where {T}
+        @assert typeof(N) <: Integer && N >= 3 "Failed constructing RegularPrismMantle with $(N) points in the polygon base (needs at least 3 points)."
+        new{N,T,typeof(z)}(r, z)
+    end
+end
+
+# Convenience functions
+const TriangularPrismMantle{T,TR} = RegularPrismMantle{3,T,TR}
+const SquarePrismMantle{T,TR}  = RegularPrismMantle{4,T,TR}
+const PentagonalPrismMantle{T,TR} = RegularPrismMantle{5,T,TR}
+const HexagonalPrismMantle{T,TR}  = RegularPrismMantle{6,T,TR}
```

Comment:
Would probably be more consistent to rename `TR` to `TZ`, but does not need to be.

Reply:
Done. Forgot to rename after copy paste